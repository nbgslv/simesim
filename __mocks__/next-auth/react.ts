const mockSession = {
  expires: new Date(Date.now() + 2 * 86400).toISOString(),
  user: {
    email: '0512345678',
    emailEmail: 'test@test.com',
    id: 'clcclcclcclcclcclc1234',
    image: null,
    name: 'foo bar',
    role: 'ADMIN',
  },
};

export const useSession = jest.fn(
  () => ({ data: mockSession, status: 'authenticated' }) // return type is [] in v3 but changed to {} in v4
);
export const signOut = jest.fn(() =>
  Promise.resolve({ url: process.env.NEXTAUTH_URL as string })
);
