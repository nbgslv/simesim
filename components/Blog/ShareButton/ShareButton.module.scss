.dropdown {
  :global .dropdown-menu {
    min-width: 8rem !important;
    width: 8rem;
  }

  :global .dropdown-item {
    padding: 0;
    font-size: 0.875rem;

    & button {
      width: 100%;
    }
  }
  :global button {
    background: transparent;
    border: none;

    &:after {
      display: none;
    }
  }
}