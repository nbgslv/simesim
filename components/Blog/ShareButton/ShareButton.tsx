import React from 'react';
import { Dropdown } from 'react-bootstrap';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { solid } from '@fortawesome/fontawesome-svg-core/import.macro';
import {
  EmailIcon,
  EmailShareButton,
  FacebookIcon,
  FacebookMessengerIcon,
  FacebookMessengerShareButton,
  FacebookShareButton,
  LinkedinIcon,
  LinkedinShareButton,
  PocketIcon,
  PocketShareButton,
  RedditIcon,
  RedditShareButton,
  TelegramIcon,
  TelegramShareButton,
  TumblrIcon,
  TumblrShareButton,
  TwitterIcon,
  TwitterShareButton,
  WhatsappIcon,
  WhatsappShareButton,
} from 'react-share';
import styles from './ShareButton.module.scss';

const ShareButton = ({ url, title }: { url: string; title: string }) => (
  <Dropdown className={styles.dropdown}>
    <Dropdown.Toggle variant="primary" id="dropdown-basic">
      <FontAwesomeIcon icon={solid('share-nodes')} />
    </Dropdown.Toggle>
    <Dropdown.Menu>
      <Dropdown.Item as="div">
        <WhatsappShareButton
          url={url}
          title={title}
          className="d-flex justify-content-between align-items-center p-2"
        >
          Whatsapp
          <WhatsappIcon style={{ width: '2.5rem', height: '2.5rem' }} />
        </WhatsappShareButton>
      </Dropdown.Item>
      <Dropdown.Item as="div">
        <FacebookShareButton
          url={url}
          className="d-flex justify-content-between align-items-center p-2"
        >
          Facebook
          <FacebookIcon style={{ width: '2.5rem', height: '2.5rem' }} />
        </FacebookShareButton>
      </Dropdown.Item>
      <Dropdown.Item as="div">
        <FacebookMessengerShareButton
          appId={process.env.NEXT_PUBLIC_FB_APP_ID as string}
          url={url}
          className="d-flex justify-content-between align-items-center p-2"
        >
          Messenger
          <FacebookMessengerIcon
            style={{ width: '2.5rem', height: '2.5rem' }}
          />
        </FacebookMessengerShareButton>
      </Dropdown.Item>
      <Dropdown.Item as="div">
        <TwitterShareButton
          url={url}
          title={title}
          className="d-flex justify-content-between align-items-center p-2"
        >
          Twitter
          <TwitterIcon style={{ width: '2.5rem', height: '2.5rem' }} />
        </TwitterShareButton>
      </Dropdown.Item>
      <Dropdown.Item as="div">
        <EmailShareButton
          url={url}
          subject={title}
          body="תראו מה מצאתי ב-שים eSim:"
          className="d-flex justify-content-between align-items-center p-2"
        >
          דוא&quot;ל
          <EmailIcon style={{ width: '2.5rem', height: '2.5rem' }} />
        </EmailShareButton>
      </Dropdown.Item>
      <Dropdown.Item as="div">
        <LinkedinShareButton
          url={url}
          title={title}
          source={url}
          className="d-flex justify-content-between align-items-center p-2"
        >
          LinkedIn
          <LinkedinIcon style={{ width: '2.5rem', height: '2.5rem' }} />
        </LinkedinShareButton>
      </Dropdown.Item>
      <Dropdown.Item as="div">
        <TelegramShareButton
          url={url}
          title={title}
          className="d-flex justify-content-between align-items-center p-2"
        >
          Telegram
          <TelegramIcon style={{ width: '2.5rem', height: '2.5rem' }} />
        </TelegramShareButton>
      </Dropdown.Item>
      <Dropdown.Item as="div">
        <RedditShareButton
          url={url}
          title={title}
          className="d-flex justify-content-between align-items-center p-2"
        >
          Reddit
          <RedditIcon style={{ width: '2.5rem', height: '2.5rem' }} />
        </RedditShareButton>
      </Dropdown.Item>
      <Dropdown.Item as="div">
        <TumblrShareButton
          url={url}
          className="d-flex justify-content-between align-items-center p-2"
        >
          Tumblr
          <TumblrIcon style={{ width: '2.5rem', height: '2.5rem' }} />
        </TumblrShareButton>
      </Dropdown.Item>
      <Dropdown.Item as="div">
        <PocketShareButton
          url={url}
          title={title}
          className="d-flex justify-content-between align-items-center p-2"
        >
          Pocket
          <PocketIcon style={{ width: '2.5rem', height: '2.5rem' }} />
        </PocketShareButton>
      </Dropdown.Item>
    </Dropdown.Menu>
  </Dropdown>
);

export default ShareButton;
