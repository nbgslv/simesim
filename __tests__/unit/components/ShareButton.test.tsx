import { render } from '@testing-library/react';
import React from 'react';
import ShareButton from '../../../components/Blog/ShareButton/ShareButton';

describe('ShareButton', () => {
  const url = 'https://example.com';
  const title = 'Example Title';

  it('renders correctly', () => {
    const { container } = render(<ShareButton url={url} title={title} />);
    expect(container).toMatchSnapshot();
  });
});
