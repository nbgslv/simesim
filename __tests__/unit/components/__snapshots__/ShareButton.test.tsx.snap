// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShareButton renders correctly 1`] = `
<div>
  <div
    aria-label="Share this blog"
    class="dropdown dropdown"
  >
    <button
      aria-expanded="false"
      class="dropdown-toggle btn btn-primary"
      id="dropdown-basic"
      type="button"
    />
  </div>
</div>
`;
